Makefile

```
@@ -1,191 +1,517 @@
 
 .PHONY: spire-server-image
-spire-server-image: Dockerfile.images ## Builds SPIRE Server docker image
-	docker build --build-arg goversion=$(goversion-required) --target spire-server -t spire-server -f Dockerfile.images .
-	# tag the image for local use (in systems tests, for example). "latest"
-	# isn't preferred since that can impact image pull policy (.e.g kubelet)
+spire-server-image: Dockerfile
+	docker build --build-arg goversion=$(go_version) --target spire-server -t spire-server .
 	docker tag spire-server:latest spire-server:latest-local
 
 .PHONY: spire-agent-image
-spire-agent-image: Dockerfile.images ## Builds SPIRE Agent docker image
-	docker build --build-arg goversion=$(goversion-required) --target spire-agent -t spire-agent -f Dockerfile.images .
-	# tag the image for local use (in systems tests, for example). "latest"
-	# isn't preferred since that can impact image pull policy (.e.g kubelet)
+spire-agent-image: Dockerfile
+	docker build --build-arg goversion=$(go_version) --target spire-agent -t spire-agent .
 	docker tag spire-agent:latest spire-agent:latest-local
 
 .PHONY: k8s-workload-registrar-image
-k8s-workload-registrar-image: Dockerfile.images ## Builds SPIRE K8S Workload Registrar docker image
-	docker build --build-arg goversion=$(goversion-required) --target k8s-workload-registrar -t k8s-workload-registrar -f Dockerfile.images .
-	# tag the image for local use (in systems tests, for example). "latest"
-	# isn't preferred since that can impact image pull policy (.e.g kubelet)
+k8s-workload-registrar-image: Dockerfile
+	docker build --build-arg goversion=$(go_version) --target k8s-workload-registrar -t k8s-workload-registrar .
 	docker tag k8s-workload-registrar:latest k8s-workload-registrar:latest-local
 
 .PHONY: oidc-discovery-provider-image
-oidc-discovery-provider-image: Dockerfile.images ## Builds SPIRE OIDC Discovery Provider docker image
-	docker build --build-arg goversion=$(goversion-required) --target oidc-discovery-provider -t oidc-discovery-provider -f Dockerfile.images .
-	# tag the image for local use (in systems tests, for example). "latest"
-	# isn't preferred since that can impact image pull policy (.e.g kubelet)
+oidc-discovery-provider-image: Dockerfile
+	docker build --build-arg goversion=$(go_version) --target oidc-discovery-provider -t oidc-discovery-provider .
 	docker tag oidc-discovery-provider:latest oidc-discovery-provider:latest-local
 
+#############################################################################
+# Code cleanliness
+#############################################################################
+
+.PHONY: tidy lint lint-code
+tidy: | go-check
+	@$(go) mod tidy
 
-##@ Others
-utils: $(utils) $(external_utils) ## Go-get SPIRE utils
+lint: lint-code
 
-$(utils): noop
-	$(docker) /bin/sh -c "go install $@"
+lint-code: $(golangci_lint_bin) | go-check
+	@PATH=$(go_bin_dir):$(PATH) $(golangci_lint_bin) run ./...
```

I've seen `golangci-lint` timing out when running in a docker container (depending on the resources assigned). Specifying a timeout between 10m to 30m shouldn't hurt and fix this problem.
```suggestion
	@PATH=$(go_bin_dir):$(PATH) $(golangci_lint_bin) --timeout 15m run ./...
```

---

The .golangci.yml file specifies a 5 minute timeout, which hasn't failed so far. I can bump it to ten though.

---

Great! I didn't realize that we had the `deadline` setting in the .yml file.